src/Umbraco.Core/Telemetry/TelemetryService.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Options;
+using Umbraco.Cms.Core.Configuration;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Manifest;
+using Umbraco.Cms.Core.Telemetry.Models;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Core.Telemetry
+{
+    /// <summary>
+    /// Service which gathers the data for telemetry reporting
+    /// </summary>
+    public sealed class TelemetryService
```

Comment:
Any reason this do not have  an interface? IMO we should introduce a `public interface ITelemetryService` and change this to internal

Reply:
I considered it, but my thought was not to create an interface to make it harder to replace in the DI container, but I guess it would be nicer to have an interface so we can make the actual implementation internal 👍

Reply:
Maybe we can just have the implementation internal, and leave it that way. I just thing methods should be virtual then

Reply:
Yeah, that's a possibility too, we'll have to move it up into infrastructure then instead of having it in core, but since it's only used in `ReportSiteTask` that shouldn't really be a problem, I just put it in core since it doesn't use anything from infrastructure

Reply:
Make sense to move it then :)